src/cover_crypt_core.rs

```diff
@@ -375,7 +396,15 @@ where
     Ok(UserSecretKey { a, b, x })
 }
 
-/// Generate the secret key encapsulation.
+/// Generates the secret key encapsulation.
+///
+/// Implements the Early Abort KEM (EAKEM) encapsulation to filter
+/// encapsulations in the decapsulation phase.
+///
+/// Encaps(pk):
+///     (c, k) ← CCA-KEM.encaps(pk)
+///     (k1, k2) ← H(k)
+///     return (k1, k2, c)
 ///
 /// # Paper
```

Comment:
do we have a link for this paper?

Reply:
Not yet, the paper is not ready for now. We can add it later on.